### PR TITLE
feat(PPDSC-2410): setup percy poc

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-2410](https://nidigitalsolutions.jira.com/browse/PPDSC-2410)

**What**

1. Background - why this is needed
https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/4108648539/PPDSC-2410+Percy+investigation
Spike to investigate how Percy could be added to the pipeline for a 2-4 week POC
2. What did you do
- Install percy dependencies
- Add scripts for Percy component and docsite tests locally and in pipeline
- Add scripts to skip Percy component and docsite tests by running single empty html test for each
- Add Percy pipeline stages alongside Applitools stages
- Skip all Percy tests that are ignored in the Applitools suite
3. What does the reviewers should expect
Percy visual tests run alongside Applitools tests

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2410-percy-pipeline-poc
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2410-percy-pipeline-poc/storybook

File: .circleci/config.yml [hunk 1]
```diff
@@ -314,6 +314,24 @@ commands:
...
+          name: Set percy/newskit-docsite token
+          command: echo "export PERCY_TOKEN=${PERCY_DOCSITE_TOKEN}" >> $BASH_ENV
+      - run:
+          name: Run Percy Docs Site Visual Tests
+          # Allow this to fail (|| true) for the POC
+          command: make e2e_visual_test_docs_percy || true
+      - store_artifacts:
+          path: cypress/screenshots
+      - store_artifacts:
+          path: cypress/videos
```
Comment (Vanals) [inline .circleci/config.yml:333]: Do we also need videos 🤔 ? Does Percy videos too?

File: .circleci/config.yml [hunk 2]
```diff
@@ -618,6 +636,24 @@ jobs:
...
           <<: *slack_notify
 
+  test_visual_comps_percy:
+    executor: node-large
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Install Headless Chrome dependencies
```
Comment (Vanals) [inline .circleci/config.yml:646]: I see we are installing these dependencies 3 times the pipeline now that we have percy. Ideally we should install this only once so we do not delay the pipeline! happy to chat about it

Comment (mstuartf) [reply .circleci/config.yml:646]: What's your preferred approach?

https://discuss.circleci.com/t/caching-apt-get-packages/9076

It looks like they recommend using a custom docker image instead of caching apt-get results, which seems... like a lot of work. Although maybe there is already one on Docker hub we could use.

Comment (Vanals) [reply .circleci/config.yml:644]: I was more thinking about installing the dependencies before these 3 jobs. In a job they have in common if the responsibility of that job can match with the task, if not to create a new job for them which install the dependencies once and attach them to the workspace!?

**Review by mutebg — DISMISSED** [review]

File: applitools.components.config.js [hunk 3]
```diff
@@ -3,14 +3,15 @@ module.exports = {
   storybookUrl: 'http://localhost:6006',
   browser: [
     {deviceName: 'iPhone 5/SE'},
-    {width: 1024, height: 768, name: 'firefox'},
-    {width: 1024, height: 768, name: 'chrome'},
-    {width: 1024, height: 768, name: 'safari'},
+    // {width: 1024, height: 768, name: 'firefox'},
+    // {width: 1024, height: 768, name: 'chrome'},
+    // {width: 1024, height: 768, name: 'safari'},
```
Comment (mutebg) [inline applitools.components.config.js:8]: Why are these commented?

File: scripts/applitools_local.sh [hunk 4]
```diff
@@ -8,39 +8,39 @@ COMMIT_SHA=$(git rev-parse HEAD)
...
+ ask the user to input their API key
+#read -p "Enter your API key (you can find this in the Applitools UI): " apiKey
 
 # the batch id must be set to the commit sha for the integration to work
 export APPLITOOLS_BATCH_ID=$(git rev-parse HEAD);
 # batch name can be anything (make it easy to identify)
 export APPLITOOLS_BATCH_NAME="LOCAL RUN by ${USERNAME} for commit ${COMMIT_SHA} on ${BRANCH_NAME}";
 # set to the user's API key
-export APPLITOOLS_API_KEY=$apiKey
+export APPLITOOLS_API_KEY=LpoCOaqCQmbArXIJLyyPcegWSPg7xXYy03101LFCrlqvA110
```
Comment (mutebg) [inline scripts/applitools_local.sh:43]: Is this supposed to be public?

Comment (mstuartf) [reply scripts/applitools_local.sh]: I committed some local changes by mistake. Good catch!

Comment (mstuartf) [reply scripts/applitools_local.sh]: I've deleted this API key from Applitools to be safe.

**Review by mutebg — APPROVED** [review]
